### DISCUSSION TITLE
Interpolating function from base mesh onto extruded mesh (i.e. interpolating from 2D mesh to 3D mesh)

### DISCUSSION THREAD

**Discussion by ethrelfall** [firedrakeproject/firedrake#3714] [Firedrake support] [answered]
Dear community,

This qn is a follow-up to my previous one.

Is there a way to interpolate from a 2D base mesh (used for a 3D extruded mesh) onto the extruded mesh such that the function is simply a constant in the extruded direction?  Also can I add include dependence on the extruded direction by applying a factor?

I've been trying to do this (using TensorProductElement with "R"-space in the extruded direction) but keep getting "geometric dimensions of source and destination meshes must match".

Apologies if I've missed something obvious ...

Search for similar done, no hits.

**Comment by colinjcotter** [comment]
This is one of those situations where it is better to ask what you are doing a bit more generally.

If you are doing stuff in a mixture of base mesh and extruded mesh it is easiest to do the base mesh stuff on the extruded mesh using function spaces that are constant in the extruded direction 
https://www.firedrakeproject.org/r-space.html#using-r-space-with-extruded-meshes
since that can be made mathematically equivalent. That way you can interpolate and project in a natural way.

**Reply by ethrelfall** [reply]
Thanks for the reply.  I'm trying to set up a basic kinetics example, something like

![image](https://github.com/user-attachments/assets/f54a3dbf-650a-436f-8d08-245a5b282f64)

The equation is 2D but the integrand in the collision term (final term on RHS) is 3D.  I want to solve the main equation in 2D (because faster) and then form the 3D integrand from the 2D solution - so I (think I) need a way of getting the 2D function onto a 3D FunctionSpace.

I guess an alternative is to somehow tell the solver that the main equation solve is really 2D so it's fast ...

Hope that makes some sense :)

**Reply by colinjcotter** [reply, marked as answer]
A bit more thought needs to be applied to the “because faster” statement. The z-independent space has the same number of DOFs as a space on the base mesh, so the linear system solve cost is the same. So there is no advantage to directly using the base mesh.

**Reply by ethrelfall** [reply]
Ah yes, I didn't set up the FunctionSpace correctly (it needed vdegree=0), now I see it's as fast as 2D - many thanks and case closed.
